api_v3/services/LiveStreamService.php

```diff
@@ -575,5 +574,97 @@ public function regenerateStreamTokenAction($entryId)
 		$entry = KalturaEntryFactory::getInstanceByType($liveEntry->getType());
 		$entry->fromObject($liveEntry, $this->getResponseProfile());
 		return $entry;
+
 	}
+
+    /**
+     *
+     * @action archive
+     * @param string $liveEntryId
+     * @return bool
+     * @throws KalturaAPIException
+     * @throws KalturaClientException
+     * @throws PropelException
+     */
+	public function archiveAction($liveEntryId)
+    {
+        /** @var LiveStreamEntry $liveEntry */
+        $liveEntry = entryPeer::retrieveByPK($liveEntryId);
+        if (!$liveEntry)
+        {
+            KalturaLog::err("Live entry id [" . $liveEntryId . "] not found");
+            return null;
+        }
+        $cuePoints = CuePointPeer::retrieveByEntryId($liveEntryId);
+
+        $importantCuePointTags = ["entry-reset-mode", "webcast:internal-state", "player-qna-settings-update",
+            "select-deck-document", "poll-data", "timeline-assets-state", "rtc-settings"];
+        /** @var CuePoint $cuePoint */
+
+        foreach ($cuePoints as $cuePoint)
+        {
+            KalturaLog::info("Iterating over cue point [" . $cuePoint->getId() . "]");
+            $cuePointTags = explode(',', $cuePoint->getTags());
+            $containsImportantTag = false;
+            foreach ($importantCuePointTags as $importantCuePointTag)
+            {
+                if (in_array($importantCuePointTag, $cuePointTags))
+                {
+                    $containsImportantTag = true;
+                    break;
+                }
+            }
+            if (!$containsImportantTag)
+            {
+                $cuePoint->setStatus(CuePointStatus::DELETED);
+                $cuePoint->save();
+            }
+        }
+        $vodEntry = entryPeer::retrieveByPK($liveEntry->getRecordedEntryId());
+        if (!$vodEntry)
+        {
+            KalturaLog::err("Recorded entry id [" . $liveEntry->getRecordedEntryId() . "] not found");
```

Comment:
If we do not have a VOD entry then we should still continue doing what we're doing since clearing also things for the live entry and clearing the push server queue are still relevant.
This does not sound like an error but maybe a warning